cmd/api/handlers.go

```diff
@@ -47,6 +48,269 @@ func (a APIServer) HandlerHealthCheck(c *gin.Context) {
+
+	c.PureJSON(http.StatusOK, NewScheduledActionListResponse(schedule))
+}
+
+// HandlerEnableScheduledAction handles the request to enable a specific scheduled action
+//
+//	@Summary		  Enables a specific Scheduled action
+//	@Description	Based on the action_id, enables the action to be executed based on its definition
+//	@Tags			    actions
+//	@Accept			  json
+//	@Produce		  json
+//	@Success		  200	{object}	nil
+//	@Failure		  500	{object}	GenericErrorResponse
+//	@Router			  /schedule [get]
+func (a APIServer) HandlerEnableScheduledAction(c *gin.Context) {
+	actionID := c.Param("action_id")
+	a.logger.Debug("Enabling action", zap.String("action_id", actionID))
+
+	err := a.sql.EnableScheduledAction(actionID)
+	if err != nil {
+		a.logger.Error("Can't enable Schedule Action", zap.String("action_id", actionID), zap.Error(err))
+		c.PureJSON(http.StatusInternalServerError, NewGenericErrorResponse(err.Error()))
+	}
+
+	c.PureJSON(http.StatusOK, nil)
+}
+
+// HandlerDisableScheduledAction handles the request to enable a specific scheduled action
+//
+//	@Summary		  Disables a specific Scheduled action
+//	@Description	Based on the action_id, disables the action to be executed based on its definition
+//	@Tags			    actions
+//	@Accept			  json
+//	@Produce		  json
+//	@Success		  200	{object}	nil
+//	@Failure		  500	{object}	GenericErrorResponse
+//	@Router			  /schedule [get]
+func (a APIServer) HandlerDisableScheduledAction(c *gin.Context) {
+	actionID := c.Param("action_id")
+	a.logger.Debug("Enabling action", zap.String("action_id", actionID))
+
+	err := a.sql.DisableScheduledAction(actionID)
+	if err != nil {
+		a.logger.Error("Can't disable Schedule Action", zap.String("action_id", actionID), zap.Error(err))
+		c.PureJSON(http.StatusInternalServerError, NewGenericErrorResponse(err.Error()))
+	}
+
+	c.PureJSON(http.StatusOK, nil)
+}
+
+// HandlerPostScheduledAction handles the request to create a new Scheduled Action
+//
+//	@Summary		  Creates a new Scheduled action
+//	@Description	Creates and registers a new scheduled action on the DB
+//	@Tags			    actions
+//	@Accept			  json
+//	@Produce		  json
+//	@Success		  200	{object}  nil
+//	@Failure		  500	{object}	GenericErrorResponse
+//	@Router			  /schedule [get]
+func (a APIServer) HandlerPostScheduledAction(c *gin.Context) {
+	a.logger.Debug("Inserting list of Scheduled Actions")
+
+	// Getting scheduled actions list on request's body
+	body, err := io.ReadAll(c.Request.Body)
+	if err != nil {
+		a.logger.Error("Can't get body from request", zap.Error(err))
+		c.PureJSON(http.StatusInternalServerError, NewGenericErrorResponse(err.Error()))
+		return
+	}
+
+	// Var for Unmarshalling results
+	var result []json.RawMessage
+
+	// Unmarshalling response
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		c.PureJSON(http.StatusInternalServerError, NewGenericErrorResponse(err.Error()))
+		return
+	}
+
+	// Unmarshalling Actions by type
+	actions, err := actions.DecodeActions(result)
```

Comment:
I guess would be better to avoid package collides

Reply:
replaced `actions` by `decodedActions`